dot/network/warp_sync_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package network
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/network/messages"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/stretchr/testify/require"
+	gomock "go.uber.org/mock/gomock"
+)
+
+func TestDecodeWarpSyncMessage(t *testing.T) {
+	t.Parallel()
+
+	// Basic WarpProofRequest
+	testWarpReqMessage := &messages.WarpProofRequest{
+		Begin: common.MustBlake2bHash([]byte("test")),
+	}
+
+	// Test encoding
+	reqEnc, err := testWarpReqMessage.Encode()
+	require.NoError(t, err)
```

Comment:
Are there any tests in substrate with hardcoded bytes that we could use here? We have run into a problem before in serialization tests where we can encode and decode to an expected type, but the actual byte representation is incorrect. 

However giving how simple this is, probably is not the end of the world. I just wanted to make a note of this

Reply:
Thanks for this suggestion Jimmy, I've been comparing this with substrate and we didn't get the same result. 
Based on that I found an issue and fix it in my last [commit](https://github.com/ChainSafe/gossamer/pull/4186/commits/f2541a37b4f6b91d4292ebebc756b95435d837fe)